bank: initial project setup

BorislavAnchev opened the pull request:
Initial project setup for the bank app task.

BorislavAnchev commented:
@mgenov, code ready for review.

mgenov commented on banktask/src/App.js:
```diff
@@ -0,0 +1,18 @@
+    <BrowserRouter>
+      <div className="App">
+        <Header data-test='Header Component'/>
+        <Route exact path='/' render={() => <Transaction />} data-testid='Transaction Page Route' />
```
HomePage sounds better than Transaction.

BorislavAnchev replied:
Changed.

BorislavAnchev replied:
Changed.

mgenov commented on banktask/src/components/Header/Header.spec.js:
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { shallow } from 'enzyme';
```
Using of react-test-renderer improves test-behaviour capabilities. Enzyme enforces coupling of test and implementation which makes code hard for maintenance.

BorislavAnchev replied:
Used `react-testing-library` as it is what you recommended on the other PR and what is recommended on the React website. Although, there may be some specific cases in which react-test-rendered is a better solution, and which I am unaware of, so I keep in mind that you may have meant use _exactly_ `react-test-renderer`.

mgenov commented on banktask/src/screens/transaction/Transaction.spec.js:
```diff
@@ -0,0 +1,223 @@
+            iban: 'BG12BUIN12341234567891',
+            currency: 'BGN',
+            balance: '5678.00',
+            history: [
```
You can lower the test data a little bit. 

You can consider and different design as when you have 1000 transactions and you go through pages you will retrieve the account information for each page.

BorislavAnchev replied:
Testing data is removed for now, as it is not required by any functionality at this moment. It was merely missed out last time. However, I will keep that in mind to couple only the needed testing data with the specific testing scenarios when such is needed.

mgenov commented on banktask/src/screens/transaction/Transaction.spec.js:
```diff
@@ -0,0 +1,223 @@
+      });
+
+      it('The submit warning should update correctly', () => {
+        const { getByTestId } = render(<Provider store={testStore({})}><MemoryRouter><Transaction /></MemoryRouter></Provider>);
```
You can improve test readability by using render wrapper function:

```jsx
const { getByTestId } = render(<Transaction />, {}) 
```

BorislavAnchev replied:
Done.

BorislavAnchev commented:
New changes ready for a review:

- Changed the name of the 'Transaction' component to 'HomeScreen'
- Swapped `Enzyme` for `react-testing-library`
- Removed the testing data as it was not used at the moment.
- Removed the wrapping <Provider> and <MemoryRouter> in the render() method to improve readability.

BorislavAnchev commented:
Added some tests for the transaction type dropdown component, but I'm not sure whether I should be writing tests for it or not.

mgenov commented on banktask/src/screens/HomePage/HomePage.spec.js:
```diff
@@ -0,0 +1,36 @@
+      expect(getByTestId('Home Page Component')).toBeInTheDocument(); // Equavalent to expect(transactionSection.length).toBe(1);
+    });
+
+    describe('Dropdown component', () => {
```
Doesn't look like it belongs here. If you are testing some external component, you are doing integration test but I'm not sure that you need this at all.

mgenov commented on banktask/src/utils/index.js:
```diff
@@ -0,0 +1,12 @@
+  return wrapper;
+};
+
+export const testStore = (initialState) => {
```
This looks really strange as it doesn't contain anything related to tests.

BorislavAnchev commented:
New changes ready for a review:

- Removed the utility functions for now as they are not needed at the moment
- Removed the tests for the dropdown component

mgenov commented on banktask/src/redux/modules/account/actions.spec.js:
```diff
@@ -0,0 +1,90 @@
+
+describe('updateAccount', () => {
+  it('updateAccount should return an object with correct type and payload', () => { 
+    const mockId = '_u70nyuzcq';
```
naming with "mock" is misleading as you are doing state testing and you are not mocking nothing actually.
